pkg/build/builder/docker.go

```diff
@@ -264,7 +264,10 @@ func (d *DockerBuilder) buildInfo() []dockerfile.KeyValue {
 // consume.
 func (d *DockerBuilder) buildLabels(dir string) []dockerfile.KeyValue {
 	labels := map[string]string{}
-	sourceInfo := d.git.GetInfo(dir)
+	sourceInfo := &s2iapi.SourceInfo{}
+	if d.build.Spec.Source.Git != nil {
```

Comment:
Yeah, with binary and git both, Dockerfile is an option.  So type is not
longer authoritative z

On Sep 30, 2015, at 8:36 AM, Ben Parees <notifications@github.com> wrote:

In pkg/build/builder/docker.go
<https://github.com/openshift/origin/pull/4804#discussion_r40762387>:

> @@ -264,7 +264,10 @@ func (d *DockerBuilder) buildInfo() []dockerfile.KeyValue {
>  // consume.
>  func (d *DockerBuilder) buildLabels(dir string) []dockerfile.KeyValue {
>  	labels := map[string]string{}
> -	sourceInfo := d.git.GetInfo(dir)
> +	sourceInfo := &s2iapi.SourceInfo{}
> +	if d.build.Spec.Source.Git != nil {

ultimately we want to move away from checking type fields and just go with
"if it's non-nil, it's intended to be used". ( @smarterclayton
<https://github.com/smarterclayton> has convinced me that's a better
approach. it's particularly useful where it might make sense to supply 2 or
more types, like gitsource+a separate dockerfile)

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4804/files#r40762387>.